src/components/App.js

```diff
@@ -1,11 +1,34 @@
+import React, { Component } from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
+import calculate from '../logic/calculate';
 
-const App = () => (
-  <>
-    <Display />
-    <ButtonPanel />
-  </>
-);
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    const newResult = calculate(this.state, buttonName);
+    this.setState(newResult);
+  }
+
+  render() {
+    const { total, next } = this.state;
+    return (
+      <div className="displayResult">
+        <Display total={next || total || '0'} />
+        <ButtonPanel clickHandler={this.handleClick} />
```

Comment:
- [ ] Your click handler is still not working, when I click any button in your calculator, nothing happens, please fix this to comply with the project requirements.